### ISSUE TITLE
Investigate the potential use of parsnp,  harvest

### ISSUE THREAD

**Issue by boulund** [ctmrbio/BACTpipe#97]
How can we best implement core genome alignment in BACTpipe? Of course, it's not something you would want to run all the time with every workflow, as it depends on the relation between the input samples you are running on. Is it reasonable to implement an optional branch in the workflow for this, and if we did, can we maintain it?

https://harvest.readthedocs.io/en/latest/

https://harvest.readthedocs.io/en/latest/content/parsnp.html

**Comment by thorellk** [comment]
I have had issues with parsnp, I filed an issue about it in the github repo some years or so ago and I don't think it has been resolved yet. Todd, who is the main developer said he would make a new release "soon" in 2017 but nothing has been updated since 2015. We originally had roary as a part of the pipeline (which also creates core genome alignments) but removed it since we thought this would not be something one would run with every workflow, just as you said. We also could not come up with a good way of flagging which genomes to include in roary so when the Nextflow version was implemented we ditched this option. But maybe we should consider it again?

**Comment by boulund** [comment]
I think it could be interesting to bring up the discussion on whether to consider having some kind of core genome alignment stuff in BACTpipe overall. Maybe this issue really has two or even three components, one of which should be brought out into its own discussion issue:

1. Do we want to include branches in BACTpipe that might not be interesting for general-purpose use?
2. How would we implement such branches in a maintainable way?
3. harvest/parsnp in the workflow (I guess this depends a bit on the outcome on the two previous issues, and also you previous experiences with it @thorellk).

**Comment by thorellk** [comment]
@boulund; I usually run Roary and then FastTree on the core genome alignment to get a quick and dirty tree. This requires all genomes in the dataset to be reasonably close though

**Comment by boulund** [comment]
Yes... Not sure if we can implement something like this in a widely-applicable way. Maybe it's best to leave to that custom downstream analyses then?

**Comment by thorellk** [comment]
Stijn Wittouck have developed a pan genome software where one doesn't need to set identity thresholds etc and that can be used on diverse genomes. Might be something?

https://github.com/SWittouck/progenomics

**Comment by boulund** [comment]
It appears to have been renamed to SCARAP
https://github.com/swittouck/scarap

It does look very interesting! Perhaps something to consider for the next major BACTpipe version?

**Comment by thorellk** [comment]
I'm still of the opinion to keep BACTpipe as simple as possible. At least when I run a core-genome analysis, I usually want to include a reference genome dataset or other genomes that are already assembled in another context, and to encompass this in BACTpipe I think is overly complex.
